doc/source/ray-air/doc_code/predictors.py

```diff
@@ -0,0 +1,83 @@
+# flake8: noqa
+# isort: skip_file
+
+# __use_predictor_start__
+import numpy as np
+import tensorflow as tf
+
+import ray
+from ray.train.batch_predictor import BatchPredictor
+from ray.train.tensorflow import (
+    to_air_checkpoint,
+    TensorflowPredictor,
+)
+
+
+def build_model() -> tf.keras.Model:
+    model = tf.keras.Sequential(
+        [
+            tf.keras.layers.InputLayer(input_shape=(1,)),
+            tf.keras.layers.Dense(1),
+        ]
+    )
+    return model
+
+
+model = build_model()
+checkpoint = to_air_checkpoint(model)
+predictor = TensorflowPredictor.from_checkpoint(
+    checkpoint, model_definition=build_model
+)
+
+data = np.array([[1], [2], [3], [4]])
+predictions = predictor.predict(data)
+print(predictions)
+# [[-1.6930283]
+#  [-3.3860567]
+#  [-5.079085 ]
+#  [-6.7721133]]
+# __use_predictor_end__
+
+# __batch_prediction_start__
+from ray.train.batch_predictor import BatchPredictor
+
+batch_predictor = BatchPredictor(
+    checkpoint, TensorflowPredictor, model_definition=build_model
+)
+predict_dataset = ray.data.range(3)
+predictions = batch_predictor.predict(predict_dataset)
+print(predictions.show())
+# {'predictions': array([0.], dtype=float32)}
+# {'predictions': array([-0.6512989], dtype=float32)}
+# {'predictions': array([-1.3025978], dtype=float32)}
+# __batch_prediction_end__
+
+# __pipelined_prediction_start__
+import pandas as pd
+import ray
+from ray.air import Checkpoint
+from ray.train.predictor import Predictor
+from ray.train.batch_predictor import BatchPredictor
+
+# Create a dummy predictor that always returns `42` for each input.
+class DummyPredictor(Predictor):
+    @classmethod
+    def from_checkpoint(cls, checkpoint, **kwargs):
+        return DummyPredictor()
+
+    def predict(self, data, **kwargs):
+        return pd.DataFrame({"a": [42] * len(data)})
+
+
+# Create a batch predictor for this dummy predictor.
+batch_pred = BatchPredictor(Checkpoint.from_dict({"x": 0}), DummyPredictor)
```

Comment:
This is really bad. Can we push on merging https://github.com/ray-project/ray/pull/26603 ?

Reply:
yeah, approved.

Reply:
with UTs addd